classes/Provider/ResetEmailerServiceProvider.php

```diff
@@ -24,12 +26,23 @@
      */
     public function register(Container $app)
     {
-        $app['reset_emailer'] = function ($app) {
+        $options = [
+            'host'     => $app->config('mail.host'),
+            'port'     => $app->config('mail.port'),
+            'username' => $app->config('mail.username'),
+            'password' => $app->config('mail.password'),
+        ];
+        $transport = (new Swift_SmtpTransport($options['host'], $options['port']))
+            ->setUsername($options['username'])
+            ->setPassword($options['password']);
+        $swiftMailer          = new Swift_Mailer($transport);
```

Comment:
As of the moment, we create all of these even if we never use the service.

How about moving all of this into the closure, and thus simplifying this method to

```php
public function register(Container $app)
{
    $app['reset_emailer'] = function ($app) {
        /** @var Application $app */
        $transport = new Swift_SmtpTransport(
            $app->config('mail.host'),
            $app->config('mail.port')
        );

        $transport
            ->setUsername($app->config('mail.username'))
            ->setPassword($app->config('mail.password'));

        return new ResetEmailer(
            new Swift_Mailer($transport),
            $app['twig'],
            $app->config('application.email'),
            $app->config('application.title')
        );
    };
}
```

Reply:
Is there a reason why `ResetEmailer` and `TalkEmailer` use separate `Swift_Mailer` instances?

Reply:
@derrabus 

Good question, I think they should share the same instance (unless there's a reason for different configuration or something).